warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java

```diff
@@ -369,51 +516,24 @@ else if (states.contains(STATE.EXECUTABLE)) {
             }
         }
         if (output != null) {
-            output.writeLine(data + node.toString() + '[' + states + (containsIgnorable ? ",IGNORABLE" : "") + "] -> " + state);
-        }
-        return state;
-    }
-    
-    /**
-     * executableUnlessItIsnt means that none of the nodes may be non-executable or error for this to be executable. Introduced to catch NULL literals in a
-     * disjuction
-     */
-    protected STATE executableUnlessItIsnt(JexlNode node, Object data) {
-        STATE state;
-        boolean containsIgnorable = false;
-        Set<STATE> states = new HashSet<>();
-        for (int i = 0; i < node.jjtGetNumChildren(); i++) {
-            states.add((STATE) (node.jjtGetChild(i).jjtAccept(this, data + PREFIX)));
-        }
-        // else remove the ignorable state
-        containsIgnorable = states.remove(STATE.IGNORABLE);
-        // if now only one state, then that is the state
-        if (states.size() == 1) {
-            state = states.iterator().next();
-        }
-        // if we contain an error state, the error
-        else if (states.contains(STATE.ERROR)) {
-            state = STATE.ERROR;
-        } else if (states.contains(STATE.NON_EXECUTABLE)) {
-            state = STATE.NON_EXECUTABLE;
-        }
-        // if we have a partial state, then partial
-        else if (states.contains(STATE.PARTIAL)) {
-            state = STATE.PARTIAL;
-        }
-        // otherwise we have an executable state
-        else {
-            return STATE.EXECUTABLE;
-        }
-        if (output != null) {
-            output.writeLine(data + node.toString() + '[' + states + (containsIgnorable ? ",IGNORABLE" : "") + "] -> " + state);
+            output.writeLine(data, node, state, false);
+            
+            // determine which states to pass up. Should be the state determined.
+            states.clear();
+            states.add(state);
+            output.pop(states);
         }
         return state;
     }
     
     @Override
     public Object visit(ASTEQNode node, Object data) {
+        if (output != null) {
+            output.disable();
+        }
+        
         STATE state;
+        Set<STATE> states = new HashSet<>();
```

Comment:
Where is this used?  Looks like it isn't.

Reply:
will remove